packages/react-components/react-dialog/src/components/DialogTitle/useDialogTitle.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { getNativeElementProps } from '@fluentui/react-utilities';
+import type { DialogTitleProps, DialogTitleState } from './DialogTitle.types';
+import { useARIAButton } from '@fluentui/react-aria';
+import { useDialogContext_unstable } from '../../contexts/dialogContext';
+import { Dismiss24Regular } from '@fluentui/react-icons';
+
+/**
+ * Create the state required to render DialogTitle.
+ *
+ * The returned state can be modified with hooks such as useDialogTitleStyles_unstable,
+ * before being passed to renderDialogTitle_unstable.
+ *
+ * @param props - props from this instance of DialogTitle
+ * @param ref - reference to root HTMLElement of DialogTitle
+ */
+export const useDialogTitle_unstable = (props: DialogTitleProps, ref: React.Ref<HTMLElement>): DialogTitleState => {
+  const { as = 'div', closeButton } = props;
+  const modalType = useDialogContext_unstable(ctx => ctx.modalType);
+
+  return {
+    components: {
+      root: 'div',
+      closeButton: 'button',
+    },
+    root: getNativeElementProps(as, {
+      ref,
+      ...props,
+    }),
+    closeButton: useARIAButton(closeButton, {
+      required: modalType === 'non-modal',
+      defaultProps: {
+        type: 'button', // This is added because the default for type is 'submit'
+        'aria-label': 'close',
```

Comment:
I guess we still really don't have a good intl solution in Fluent. but apps will have to intl their labels themselves. If this was on the root slot I don't think it's that bad, but the fact that accessing this requires:

```tsx
<DialogTitle closeButton={{ 'aria-label': 'schließen' }} />
```

seems kind of awkward to me

Reply:
Yeah, I was worried about that also... we need to find a better way to add labels internally.